src/backend/gl/src/device.rs

```diff
@@ -1713,7 +1714,146 @@ impl d::Device<B> for Device {
         config: c::SwapchainConfig,
         _old_swapchain: Option<Swapchain>,
     ) -> Result<(Swapchain, Vec<n::Image>), c::window::CreationError> {
-        Ok(self.create_swapchain_impl(surface, config))
+        let gl = &self.share.context;
+
+        let (int_format, iformat, itype) = match config.format {
+            Format::Rgba8Unorm => (gl::RGBA8, gl::RGBA, gl::UNSIGNED_BYTE),
+            Format::Bgra8Unorm => (gl::RGBA8, gl::BGRA, gl::UNSIGNED_BYTE),
+            Format::Rgba8Srgb => (gl::SRGB8_ALPHA8, gl::RGBA, gl::UNSIGNED_BYTE),
+            _ => unimplemented!(),
+        };
+
+        let channel = config.format.base_format().1;
+
+        #[cfg(feature = "wgl")]
+        let context = {
+            use crate::window::wgl::PresentContext;
+
+            let context = PresentContext::new(surface, &self.share.instance_ctxt);
+            context.make_current();
+            context
+        };
+
+        let mut fbos = Vec::new();
+        let mut images = Vec::new();
+
+        for _ in 0..config.image_count {
+            unsafe {
+                let mut fbo = 0;
+                gl.GenFramebuffers(1, &mut fbo);
+                gl.BindFramebuffer(gl::FRAMEBUFFER, fbo);
+                fbos.push(fbo);
+
+                let image = if config.image_layers > 1
+                    || config.image_usage.contains(i::Usage::STORAGE)
+                    || config.image_usage.contains(i::Usage::SAMPLED)
+                {
+                    let mut name = 0;
+                    gl.GenTextures(1, &mut name);
+                    match config.extent {
+                        Extent2D {
+                            width: w,
+                            height: h,
+                        } => {
+                            gl.BindTexture(gl::TEXTURE_2D, name);
+                            if self.share.private_caps.image_storage {
+                                gl.TexStorage2D(
+                                    gl::TEXTURE_2D,
+                                    config.image_layers as _,
+                                    int_format,
+                                    w as _,
+                                    h as _,
+                                );
+                            } else {
+                                gl.TexParameteri(
+                                    gl::TEXTURE_2D,
+                                    gl::TEXTURE_MAX_LEVEL,
+                                    (config.image_layers - 1) as _,
```

Comment:
why are we treating `image_layers` as the mipmap count and allocating accordingly?

Reply:
Appears to be a bug in the current implementation, fixed it. Nice catch!